libs/hdf-converters/src/veracode-mapper.ts

```diff
@@ -0,0 +1,552 @@
+/* eslint-disable prettier/prettier */
+import { integer } from 'aws-sdk/clients/cloudfront';
+import parser from 'fast-xml-parser';
+import {ExecJSON} from 'inspecjs';
+import _ from 'lodash';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {
+  BaseConverter,
+  ILookupPath,
+  MappedTransform
+} from './base-converter';
+import {CweNistMapping} from './mappings/CweNistMapping';
+const CWE_LENGTH = 'cwe.length'
+const SCA_VULNERABILITIES = 'detailedreport.software_composition_analysis.vulnerabilities['
+const REPORT_SEVERITY = 'detailedreport.severity['
+const SCA_CVES = 'detailedreport.software_composition_analysis.cves['
+const CATEGORY = '].category['
+const CWE_NIST_MAPPING = new CweNistMapping();
+const DEFAULT_NIST_TAG = ['SI-2', 'RA-5'];
+const IMPACT_MAPPING: Map<string, number> = new Map([
+  ['5', 0.9],
+  ['4', 0.7],
+  ['3', 0.5],
+  ['2', 0.3],
+  ['1', 0.1],
+  ['0', 0.0]
+]);
+
+function impactMapping(severity: number): number {
+  if (typeof severity === 'string' || typeof severity === 'number') {
+    return IMPACT_MAPPING.get(severity.toString().toLowerCase()) || 0.1;
+  } else {
+    return 0.1;
+  }
+}
+
+function nistTag(input: string): string[] {
+  const len = Number(`${_.get(input, CWE_LENGTH)}`)
+  const cwes = []
+  for (let index = 0; index < len; index++) {
+    cwes.push(`${_.get(input, 'cwe.cweid')}`)
```

Comment:
i'd appreciate it if you could write out to a file the json file that was derived from converting it from xml (not the hdf file) and sending it to me over slack.

anyways does not seem to be incorporating the index into which cweid it's getting which i think also probably explains why it's getting the default nist tags every time (iirc) cause cwe.cweid is not a thing - think you'd need cwe[index].cweid

anyways, instead of using for loops, we could probably use a map here - something like `const cwes = _.get(input, 'cwe').map(({cweid}) => cweid);`

Reply:
fixed